src/transformers/trainer.py

```diff
@@ -2011,7 +2011,7 @@ def _inner_training_loop(
                         ):
                             grad_norm = model.get_global_grad_norm()
                         else:
-                            grad_norm = _grad_norm.item() if _grad_norm is not None else None
+                            grad_norm = _grad_norm  # Keep grad_norm as a tensor to avoid synchronization of TPUs through .item().
```

Comment:
great catch !

Reply:
Can we do this only if we're on a TPU? Otherwise we'll eventually get back to this issue: https://github.com/huggingface/transformers/issues/29207, which is working on being solved in this PR: https://github.com/huggingface/transformers/pull/29212